### ISSUE TITLE
Interfacing EPICS in NICOS

### ISSUE THREAD

**Issue by nikhilbiyani** [DMSC-Instrument-Data/ecp-backlog#1]
The current interface is based on the use of caget/caput style calls with timeouts. This may cause NICOS to become unresponsive when a IOC goes away. The NICOS EPICS interface needs to be rewritten to use the monitor interface, in pyepics speak: Automatic Monitoring of a PV. This will call automatically a callback when a PV changes value, disconnects or goes into an alarm state.

**Comment by MikeHart85** [comment]
Are we still using pyepics, or have we switch to pvaPy already?

There seem to be two PRs from Michael Wedel that switch to pvaPy that are merged already, here:
https://forge.frm2.tum.de/review/#/c/15350/
https://forge.frm2.tum.de/review/#/c/15352/

If this is still using pyepics, could it be that this issue will go away (or become a different issue) when we switch to pvaPy. According to the [notes from the ECP workshop](https://confluence.esss.lu.se/display/DMSC/5th+ESS+Controls+Workshop%2C+19th+-+21st%2C+PSI), we will definitely have to make that switch.

**Comment by nikhilbiyani** [comment]
Yes, eventually we will use pvaPy. But the current pvaPy implementation is not stable, leads to some segfaults which one needs to investigate. This might be a problem from pvaPy. Also, for now, the sinq-amorsim is not based on EPICS 4, so using pv access is not possible for the devices.

**Comment by mkoennecke** [comment]
Well,,

PvaPy can speak CA, so the real reason is the stability issue with pvaPy.

**Comment by nikhilbiyani** [comment]
One more solution is to use services like `forward-epics-to-kafka` to receive PV updates and a similar service to request PV changes. Then the dependency on `pyepics` or `pvapy` will be removed.

**Comment by mattclarke** [comment]
Issue moved to JIRA https://jira.esss.lu.se/browse/DM-695
